### PR TITLE
Disable some tests for Windows IoT

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/34051
Fix https://github.com/dotnet/corefx/issues/34024
Fix https://github.com/dotnet/corefx/issues/33982

All tools or binaries or entrypoints not existing on IoT SKU.

### CODE REVIEW

File: src/System.DirectoryServices.AccountManagement/tests/PrincipalContextTests.cs [hunk 1]
```diff
@@ -288,7 +288,7 @@ public void UserName_GetWhenDisposed_ThrowsObjectDisposedException()
             Assert.Throws<ObjectDisposedException>(() => context.UserName);
         }
 
-        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer), nameof(PlatformDetection.IsNotWindowsIoTCore)))]
```
Comment (stephentoub) [inline src/System.DirectoryServices.AccountManagement/tests/PrincipalContextTests.cs]: Why did you change this to ConditionalFact?  How does that work with arguments and InlineDatas?

Comment (danmosemsft) [reply src/System.DirectoryServices.AccountManagement/tests/PrincipalContextTests.cs]: Trying to do two things at once.
